src/pydicom/dataset.py

```diff
@@ -2656,6 +2660,56 @@ def ensure_file_meta(self) -> None:
         if not hasattr(self, "file_meta"):
             self.file_meta = FileMetaDataset()
 
+    def validate(self, _init_value: _DatasetType) -> None:
+        """Validate the dataset.
+
+        Performs the same validation checks as dcmwrite without writing to disk.
+
+        Parameters
+        ----------
+            _init_value : _DatasetType
+                Unused, required for compatibility with FileMetaDataset.
+
+        Returns
+        -------
+        None
+
+
+        Raises
+        ------
+        ValueError
+            If the dataset is invalid.
+        """
+        # Check for disallowed tags
+        bad_tags = [x >> 16 for x in self._dict if x >> 16 in (0, 2)]
+        if bad_tags:
+            if 0 in bad_tags:
+                raise ValueError(
+                    "Command Set elements (0000,eeee) are not allowed in datasets"
+                )
+            else:
+                raise ValueError(
+                    "File Meta Information Group elements (0002,eeee) must be in a "
+                    "FileMetaDataset instance in the 'file_meta' attribute"
+                )
+
+        # Validate File Meta Information if present
+        if hasattr(self, "file_meta"):
+            validate_file_meta(self.file_meta, enforce_standard=True)
+
+        # Check TransferSyntaxUID if present
+        if "TransferSyntaxUID" in self.file_meta:
+            tsyntax = self.file_meta.TransferSyntaxUID
+            if tsyntax and not tsyntax.is_private and tsyntax.is_transfer_syntax:
+                if "PixelData" in self:
+                    self["PixelData"].is_undefined_length = tsyntax.is_compressed
+
+        # Validate preamble if present
+        if hasattr(self, "preamble"):
+            preamble = self.preamble
+            if preamble and len(preamble) != 128:
+                raise ValueError("'preamble' must be 128-bytes long")
```

Comment:
Only required if writing to the DICOM File Format. A dataset can be valid without a preamble.